packages/dds/merge-tree/docs/IntervalStickiness.md

```diff
@@ -0,0 +1,158 @@
+ ^^^+
+```
+
+This works today because the newly inserted segment _does_ actually fall within the bounds of the interval, and after insertion, the end bound of the right interval slides to the next farthest segment, expanding the interval to `[1, 4)` or `[bcXd)`.
+
+This same trick does work for left-stickiness when we only consider insertion, but breaks down in the case of removal. An example of this is:
+
+If we take the diagram of the range `(1, 3]`
+
+```
+abcdef
+ +^^
+```
+
+and we delete the character `b`, the interval bound today will slide to the right, shrinking the interval.
+
+```
+acdef
+ +^^
+```
+
+The expected behavior would be for the bound to slide to the character `a`.
+
+#### "Left" and "Right"
+
+"Left" and "right" in the context of text can be confusing, as some scripts are written right-to-left. For the purposes of this document, left/forward/near are all synonyms referring to text that is closer to the start of the string, and right/backward/far are synonyms referring to text that is closer to the end of the string.
+
+### Detailed Design
+
+The implementation work is largely straightforward. In order to get the correct left-sticky behavior in the face of removal, we just have to modify local reference sliding to prefer sliding to the left when the reference is the start-bound of a left-sticky interval. Just changing the order of traversal when looking for slid-to-segments is sufficient to implement this change and make left-stickiness tests pass.
+
+#### String Endpoints
+
+Exclusive bounds raise the problem of spanning over the start and end characters of the string. That is, if we convert an inclusive range `[0, 1]` to an exclusive, left-sticky range, the leftmost range bound must become negative: `(-1, 1]`. This is not supported today. The proposed solution is to have a marker segment that always lives at the start of the string.
+
+There is prior art here in existing implementations of right-stickiness that insert a marker segment that always lives at the end of the string.
+
+This document proposes to make both start-of-string and end-of-string markers that can be referred to by intervals a first class concept. An `EndOfTreeSegment` structure exists in `merge-tree` today, and can be used to implement such an end-of-string marker for right-sticky intervals. A similar structure should be created for `StartOfTreeSegment`.
+
+#### Stickiness as a First Class Concept
+
+Although it is possible to implement right-stickiness today, doing so requires manual tracking of whether a bound is exclusive or inclusive. With the addition of leftmost exclusive bounds to support full- and left- stickiness, this tracking becomes significantly more complex for the end user.
+
+This document proposes adding first class support for declaring intervals as left/right/full/not sticky. This can be achieved by adding a field to intervals that tracks the interval stickiness and updating some methods to be aware of this new functionality.
```

Comment:
we'll need to serialize things appropriately too. I'd mention plans for test coverage & serialized format in this realm.